extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_2_speech.py

```diff
@@ -0,0 +1,240 @@
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [_pipeline_creation_params, unfiltered_completion_params] = refine_pipeline_creation_params(model_settings)
+        completion_data = refine_completion_params(unfiltered_completion_params)
+
+        # Add resolved prompt
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+        completion_data["prompt"] = resolved_prompt
+        return completion_data
+
+    async def run_inference(self, prompt: Prompt, aiconfig: "AIConfigRuntime", options: InferenceOptions, parameters: Dict[str, Any]) -> List[Output]:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            InferenceResponse: The response from the model.
+        """
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [pipeline_creation_data, _] = refine_pipeline_creation_params(model_settings)
+        if not pipeline_creation_data.get("requires_safety_checker", True):
+            pipeline_creation_data["safety_checker"] = None
+
+        model_name: str = aiconfig.get_model_name(prompt)
+        if isinstance(model_name, str) and model_name not in self.synthesizers:
+            self.synthesizers[model_name] = pipeline("text-to-speech", model_name)
+        synthesizer = self.synthesizers[model_name]
+
+        completion_data = await self.deserialize(prompt, aiconfig, options, parameters)
+        inputs = completion_data.pop("prompt", None)
+        response = synthesizer(inputs, **completion_data)
+
+        outputs: List[Output] = []
+        assert not isinstance(response, list)
+        for count, audio in enumerate([response]):
+            output = construct_output(audio, count)
+            outputs.append(output)
+
+        prompt.outputs = outputs
+        return prompt.outputs
+
+    def get_output_text(
+        self,
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        output: Optional[Output] = None,
+    ) -> str:
+        if output is None:
+            output = aiconfig.get_latest_output(prompt)
+
+        if output is None:
+            return ""
+
+        # TODO (rossdanlm): Handle multiple outputs in list
+        # https://github.com/lastmile-ai/aiconfig/issues/467
+        if output.output_type == "execute_result":
+            if isinstance(output.data, OutputDataWithValue):
+                return output.data.value
+            elif isinstance(output.data, str):
+                return output.data
+        return ""
+
+    def _get_device(self) -> str:
+        if torch.cuda.is_available():
+            return "cuda"
+        elif torch.backends.mps.is_available():
+            return "mps"
+        return "cpu"
```

Comment:
If not used let's delete. Does the generator not take a `toDevice()` field to make this more efficient? https://github.com/lastmile-ai/aiconfig/blob/54de9cc8a1055975b4d28b877bea3dbdc627644e/extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_2_image.py#L290-L294